spec/services/discovery_engine/quality/sample_query_set_spec.rb

```diff
@@ -74,6 +75,28 @@
             .with("SampleQuerySet clickstream 2025-10 already exists. Skipping query set creation...")
         end
       end
+
+      context "when the sample query set is not created" do
+        let(:erroring_service) { double("sample_query_set") }
+
+        before do
+          allow(DiscoveryEngine::Clients).to receive(:sample_query_set_service).and_return(erroring_service)
+          allow(erroring_service).to receive(:create_sample_query_set).with(anything)
+
+          allow(erroring_service)
+            .to receive(:get_sample_query_set)
+            .with(anything)
```

Comment:
You could maybe pass in the real arguments here instead of `anything` (ie the `name`). I'm not an expert on correct use of the `anything` matcher but I think it's probably a useful tool if we're stubbing a "broken" endpoint - something that just returns an error however we interact with it. But in this case we're only returning the 404 for this specific sample query set name.